### ISSUE TITLE
Simplify EnumVariant

### ISSUE THREAD

**Issue by dtolnay** [joshleeb/varianteq#2]
The `{..}` pattern syntax is accepted for any struct or variant type including unit variants and tuple variants. This is specifically to make this sort of pattern matching easier where we do not care about the variant type or fields. So EnumVariant can be implemented as:

```rust
let enum_ident = self.enum_ident;
let variant_ident = self.variant_ident;

tokens.append_all(quote! {
    (#enum_ident::#variant_ident{..}, #enum_ident::#variant_ident{..})
});
```

This expands like:

```rust
enum E {
    A,
    B(i32, i32),
    C { x: i32 },
}

impl PartialEq for E {
    fn eq(&self, other: &E) -> bool {
        match (self, other) {
            (E::A{..}, E::A{..}) => true,
            (E::B{..}, E::B{..}) => true,
            (E::C{..}, E::C{..}) => true,
            _ => false,
        }
    }
}
```

**Comment by joshleeb** [comment]
Thanks @dtolnay 🥇 I didn't know that was possible in Rust. I'll change the implementation to take advantage of this shortly 😄

**Comment by joshleeb** [comment]
Resolved by #3
